alphaVantage/alphavantage.js

```diff
@@ -0,0 +1,98 @@
+const axios = require("axios");
+const utils = require("./utils");
+
+module.exports = class alphaVantage {
+	constructor(key) {
+		this.key = key;
+	}
+
+	getStockQuote = async (symbol) => {
+		let uri = `https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=${symbol}&apikey=${this.key}`;
+		let data = await axios
+			.get(uri)
+			.then((res) => {
+				return res.data["Global Quote"];
+			})
+			.catch((err) => {
+				console.log(err);
+				return {};
+			});
+
+		if ("05. price" in data) {
+			const price = data["05. price"];
+			return price;
+		} else {
+			return 0;
+		}
+	};
+
+	getStockInfo = async (keyword) => {
+		let uri = `https://www.alphavantage.co/query?function=SYMBOL_SEARCH&keywords=${keyword}&apikey=${this.key}`;
+		let data = await axios
+			.get(uri)
+			.then((res) => {
+				return res.data["bestMatches"][0];
+			})
+			.catch((err) => {
+				console.log(err);
+				return {};
+			});
+
+		if ("1. symbol" in data) {
+			const info = { symbol: data["1. symbol"], name: data["2. name"] };
+			return info;
+		} else {
+			return "No data found";
+		}
+	};
+
+	// The whole command to get data regarding a stock for the frontend
+	getStockData = async (symbol) => {
+		let uri = `https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY&symbol=${symbol}&interval=5min&outputsize=full&apikey=${this.key}`;
+		let timeCoordinates = await axios
+			.get(uri)
+			.then((res) => {
```

Comment:
I would probably use another variable and `await` instead of chaining `.then` to reduce nesting, then use `try`/`catch` for error handling.

Reply:
Actually, now that I think about it, you would have to rebind data (which might look messier?), so I'm gonna resolve this.